lib/tasks/asset_sync.rake

```diff
@@ -20,11 +20,24 @@ if Rake::Task.task_defined?("assets:precompile:nondigest")
     Rake::Task["assets:sync"].invoke if defined?(AssetSync) && AssetSync.config.run_on_precompile
   end
 else
+  # Triggers on webpacker compile task instead of assets:precompile to ensure
```

Comment:
Ah, I think I understand what's going on here. I thought your simpler version would work, but I think the implementation ~~we stole~~ that inspired this fork is probably better.

The intention is to ensure that we enhance either the "webpacker:compile" or "assets:precompile" tasks but never both. I have no idea why someone would have the webpacker gem installed and then say "nah, don't sync that like all my other assets," but here we are.